metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -492,11 +492,15 @@ def close(self) -> None:
 
     def _infer_platform(self) -> Optional[str]:
         config = self.get_config()
-        return (
+        platform = (
             getattr(config, "platform_name", None)
             or getattr(self, "platform", None)
             or getattr(config, "platform", None)
         )
+        if platform is None and hasattr(self, "get_platform_id"):
+            platform = type(self).get_platform_id()
```

Comment:
get_platform_id comes from the decorators, right?

Reply:
Yes. This was observed for dremio that platform_instance entry was not getting added in browse path v2.